### PR TITLE
frontend: make navbar resizable

### PR DESCRIPTION
**Pull request by dameekoh**
I made the project navbar resizable on hover similarly to that of the supabase.

when project navbar is hovered, avatarmenu also displays the username.

image
image
image
image

----

> [!IMPORTANT]
> Replaces collapsed navbar with a resizable version and updates avatar menu to show username on hover.
> 
>   - **Behavior**:
>     - Replaces `ProjectNavbarCollapsed` with `ProjectNavbarResizable` in `layout.tsx` to make the navbar resizable on hover.
>     - Updates `AvatarMenu` to display the username when the navbar is hovered.
>   - **Components**:
>     - Deletes `project-navbar-collapsed.tsx`.
>     - Adds `project-navbar-resizable.tsx` with hover functionality to expand the navbar width and show option names.
>     - Modifies `avatar-menu.tsx` to include username display on hover.
> 
> <sup>This description was created by </sup>[Ellipsis](https://www.ellipsis.dev?ref=lmnr-ai%2Flmnr&utm_source=github&utm_medium=referral)<sup> for 1ccee10f69744af08f7577ab5846fabeba0016b6. It will automatically update as commits are pushed.</sup>

### CODE REVIEW

File: frontend/components/project/project-navbar-resizable.tsx [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+  });
+
+  return (
+    <div className="flex flex-col h-screen border-r text-md items-start w-14 relative group">
+      <div className="absolute top-0 left-0 h-full hover:w-48 w-14 transition-[width] duration-200 ease-out border-r z-50 bg-background">
+        <Link
+          href={'/projects'}
+          className="flex h-14 items-center w-14 group-hover:w-full px-4 py-2"
+        >
+          <Image alt="Laminar AI icon" src={logo} height={20} />
```
Comment (skull8888888) [inline frontend/components/project/project-navbar-resizable.tsx]: Let's switch to wide logo (which is used on the projects page) when sidebar is extended

File: frontend/components/user/avatar-menu.tsx [hunk 2]
```diff
@@ -12,19 +12,24 @@ import { useUserContext } from '@/contexts/user-context';
...
-            src={imageUrl}
-            alt="avatar"
-            width={28}
-            height={28}
-            className="border rounded-full cursor-pointer"
-          />
+        <div className="flex items-center space-x-2">
+          <div className="h-7 w-7 overflow-hidden rounded-full border">
+            <Image
+              src={imageUrl}
```
Comment (skull8888888) [inline frontend/components/user/avatar-menu.tsx]: Let's show first letter of the username and random color if imageUrl is null or undefined

**Review by skull8888888 — CHANGES_REQUESTED** [review]
Overall looks good, added some comments, and can you please remove diff on lock file.

**Comment by dinmukhamedm** [comment]
Also @dameekoh please change the destination (target) branch to `dev`

File: frontend/components/user/avatar-menu.tsx (hunk 2 above)
Comment (dameekoh) [reply frontend/components/user/avatar-menu.tsx]: I refactored, check it out. However, it seems that the icon on the expanded looks smaller than collapsed. I suppose it's diminished in the design svg file

**Comment by dameekoh** [comment]
хз чето не получилось pnpm-lock.yaml удалить. но он в принципе обычно коммитится для reproducable builds.
{3644BE85-AA44-4902-AAF4-11D6D4305088}

**Comment by skull8888888** [comment]
> хз чето не получилось pnpm-lock.yaml удалить. но он в принципе обычно коммитится для reproducable builds. {3644BE85-AA44-4902-AAF4-11D6D4305088}

I meant remove the **diff** on the lock file, i.e. keep it as it was before

**Pull request closed without merging**